docs/diagrams/ui-class-diagram.puml

```diff
@@ -0,0 +1,61 @@
+@startuml
+hide circle
+skinparam classAttributeIconSize 0
+
+class Ui {
+
+    {static} + COLUMN_WIDTH: int = 10
+    {static} + LIST_COLUMN_WIDTH: int = 30
+    {static} - ELLIPSIS: String = "..."
+    {static} - PROGRAM_NAME: String = "FinText"
+    {static} - FILLER_CHAR: char = ' '
+    {static} - LIST_SEPARATOR: char = '='
+    {static} - ID_COLUMN_WIDTH: int = 3
+    {static} - SPACE_BETWEEN_COLS: int = 3
+    {static} - AMOUNT_FORMAT: String = "%.2f"
+    {static} - LINE_DELIMITER: char = '\\n'
+
+    + Ui()
+    + Ui(outputstream: OutputStream)
+    + printTableRow(rowValues: ArrayList<String>): void
+    + printTableRow(rowValues: ArrayList<String>, headers: String[]): void
+    + printTableRow(rowValues: ArrayList<String>, headers: String[], customWidths: Integer[]): void
+    + printTableRows(rows: ArrayList<ArrayList<String>>): void
+    + printTableRows(rows: ArrayList<ArrayList<String>>, headers: String[]): void
+    + printTableRows(rows: ArrayList<ArrayList<String>>, headers: String[], customWidths: Integer[]): void
+    + printTableHeader(headers: String[], customWidths: Integer[])
+    + formatAmount(value: Double): String
+    + print(value: String): String
+    + printGreeting(): String
+    + printBye(): String
+    + readUserInput(): String
+    + close(): void
```

Comment:
You might want to keep your diagrams simple and omit unimportant and private members 
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/93811746/12abef0a-0010-42ea-ad39-c862fdca9626)